gridData/core.py

```diff
@@ -257,12 +257,32 @@ def resample_factor(self, factor):
         --------
         resample
 
+        .. versionchanged:: 0.6.0
+           Previous implementations would not alter the range of the grid edges
+           being resampled on. As a result, values at the grid edges would creep
+           steadily inward. The new implementation recalculates the extent of
+           grid edges for every resampling.
+
         """
-        # new number of edges N' = (N-1)*f + 1
-        newlengths = [(N - 1) * float(factor) + 1 for N in self._len_edges()]
-        edges = [numpy.linspace(start, stop, num=int(N), endpoint=True)
-                 for (start, stop, N) in
-                 zip(self._min_edges(), self._max_edges(), newlengths)]
+        if float(factor) <= 0:
+            ValueError("Factor must be positive")
+        # Determine current spacing
+        spacing = (numpy.array(self._max_edges(
+        )) - numpy.array(self._min_edges())) / (-1 + numpy.array(self._len_edges()))
```

Comment:
Formatting is a bit odd, maybe
```suggestion
        spacing = (numpy.array(self._max_edges()) \
                   - numpy.array(self._min_edges())) / \
                  (-1 + numpy.array(self._len_edges())
```

Reply:
... sometimes just ignoring PEP8 is also acceptable ;-)